perception/tensorrt_yolox/CMakeLists.txt

```diff
@@ -0,0 +1,189 @@
+cmake_minimum_required(VERSION 3.5)
+project(tensorrt_yolox)
+
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 17)
+  set(CMAKE_CXX_STANDARD_REQUIRED ON)
+  set(CMAKE_CXX_EXTENSIONS OFF)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic -Werror)
+endif()
```

Comment:
It is better to use autoware_cmake
https://github.com/autowarefoundation/autoware.universe/blob/main/perception/multi_object_tracker/CMakeLists.txt#L4-L5

Reply:
@yukkysaito 
Thank you very much for the comment. I replaced the lines you mentioned with the macro provided by `autoware_cmake` in [this commit](a314e8d791111645d68057deefcfbb8147d5dab7).
I would appreciate it if you check that my modification meets your intention.

Reply:
@manato LGTM